README.md

```diff
@@ -1,2 +1,665 @@
+
+🎁  We have added the possibility to use the following search parameters `attributesToHighlight`, `attributesToCrop`, `cropLength` as settings (see example above).
+
+
+## 🔍 Custom search
+
+All the supported options are described in the [search parameters](https://docs.meilisearch.com/reference/features/search_parameters.html) section of the documentation.
+
+```ruby
+Book.search('Harry', { filters: 'author = J. K. Rowling' })
+```
+👉 Don't forget that `attributesToHighlight`, `attributesToCrop` and `cropLength` can be set up as settings in the MeiliSearch block of your model.
+
+
+## 🪛 Options
+
+### MeiliSearch configuration & environment
+
+#### Custom index_uid
+
+By default, the **index_uid** will be the class name, e.g. `Book`. You can customize the index_uid by using the `index_uid` option
+
+```ruby
+class Book < ActiveRecord::Base
+  include MeiliSearch
+  meilisearch :index_uid => 'MyCustomUID' do
+  end
+end
+```
+
+#### Per-environment index_uid
+
+You can suffix the index_uid with the current Rails environment using the following option:
+
+```ruby
+class Book < ActiveRecord::Base
+  include MeiliSearch
+  meilisearch per_environment: true do # index name will be "Book_#{Rails.env}"
+  end
+end
+```
+
+### Index configuration
+
+#### Custom attribute definition
+
+You can add a custom attribute by using the `add_attribute` option or by using a block.
+
+⚠️ When using custom attributes, the gem is not able to detect changes on them. Your record will be pushed to the API even if the custom attribute didn't change. To prevent this behavior, you can create a `will_save_change_to_#{attr_name}?` method.
+
+```ruby
+class Author < ApplicationRecord
+  include MeiliSearch
+
+  meilisearch do
+    attribute :first_name, :last_name
+    attribute :full_name do
+      '#{first_name} #{last_name}'
+    end
+    add_attribute :full_name_reversed
+  end
+
+  def full_name_reversed
+    '#{last_name} #{first_name}'
+  end
+
+  def will_save_change_to_full_name?
+    will_save_change_to_first_name? || will_save_change_to_last_name?
+  end
+
+  def will_save_change_to_full_name_reversed?
+    will_save_change_to_first_name? || will_save_change_to_last_name?
+  end
+end
+```
+
+#### Custom primary key
+By default, the `primary key` is based on your record's id. You can change this behavior specifying the `:primary_key` option.
+
+Note that the primary key must have a **unique value**.
+
+```ruby
+class Book < ActiveRecord::Base
+  include MeiliSearch
+  meilisearch :primary_key => 'ISBN' do
+  end
+end
+```
+#### Conditional indexing
+
+You can control if a record must be indexed by using the `:if` or `:unless` options
+As soon as you use those constraints, add_documents and delete_documents calls will be performed in order to keep the index synced with the DB. To prevent this behavior, you can create a `will_save_change_to_#{attr_name}?` method.
```

Comment:
```suggestion
As soon as you use those constraints, `add_documents` and `delete_documents` calls will be performed in order to keep the index synced with the DB. To prevent this behavior, you can create a `will_save_change_to_#{attr_name}?` method.
```